editor/src/components/editor/store/dispatch.tsx

```diff
@@ -94,11 +95,14 @@ import { isFeatureEnabled } from '../../../utils/feature-switches'
 import { isJsOrTsFile, isCssFile } from '../../../core/shared/file-utils'
 import { applyStatePatches } from '../../canvas/commands/commands'
 
-export interface DispatchResult extends EditorStore {
+interface DispatchResultFields {
   nothingChanged: boolean
   entireUpdateFinished: Promise<any>
 }
 
+export interface InnerDispatchResult extends EditorStoreExplicit, DispatchResultFields {}
```

Comment:
as we talked about this previously, I think this `InnerDispatchResult` could Omit<'patchedEditor'> to make sure that the code around processing actions is not even aware of the state patching